plugins/focus-tracker/index.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Allows users to see what screen readers would see.
+ */
+
+let Plugin = require("../base");
+
+let annotate = require("../shared/annotate")("tabIndex");
+
+const FOCUS_EVENT = "focusin";
+const BLUR_EVENT = "focusout";
+
+// this will let us get a shorter info panel that just
+// lets the user know we are tracking their focus
+const PANEL_OPTIONS = {
+  statusPanelView: true
+};
+
+// we're going to use focusin and focusout because they bubble
+const FOCUS_STATES = {
+  [FOCUS_EVENT]: "tota11y-outlined",
+  [BLUR_EVENT]: "tota11y-was-focused"
+};
+
+// we'll use this to make sure we don't apply the was-focused
+// indicator to our tota11y panels
+const IGNORE = "tota11y"
+
+// we're going to attempt to visualize the tab order of the page
+// based on the source order of the tabbable elements
+// this Array contains our tabbable element selectors
+const TABABLE_ELEMENTS = [
+    "a[href]",
+    "area[href]",
+    "iframe",
+    "input:not([disabled])",
+    "select:not([disabled])",
+    "textarea:not([disabled])",
+    "button:not([disabled])",
+    "[tabIndex]:not([tabIndex=\"-1\"])",
```

Comment:
So this catches anything that is "made" tabbable by adding a non-negative tabindex? Makes sense.

I wonder, could you comment on this and the contenteditable one to explain what they do? I think it will be really useful for folks looking at this code later.

Reply:
I'll add more details and some references to where this list came from.